### PR TITLE
Add debug target to avoid always stripping debug symbols

### PR DESCRIPTION
**Pull request by rtorrero**
This PR adds a new debug target which behaves slightly different from the `default` target:

 - It does not run the tests
 - It doesn't use `-trimpath` which causes some issues while debugging remotely with `delve`
 - It doesn't use `-s -w` in the `LDFLAGS` which cause thinner binaries but strips symbols away needed for debugging

Note that this is only useful for development and should provide identical binaries as before when running the `default` target.

### CODE REVIEW

**Review by stefanotorresi — COMMENTED** [review]
I'm still not sure that an additional make target is the way to go. The basic target to compile is `make build`. I think we should just make that more flexible.
Have you considered something like `DEBUG=true make`? This way the build flags would be changed no matter what target the user is selecting, be that `make build` or `make cross-compile` or whatever: otherwise, one is forced to run all the things that are in the new `debug` target, and is left with no ability to fine-tune the build flags when using all the other targets.

**Comment by rtorrero** [comment]
Yes, now that I think about it it makes much more sense to use a conditional in this case. Let me know if it's what you meant; I'll squash & merge :+1:

**Comment by arbulu89** [comment]
@stefanotorresi I think this is waiting for you. Honestly, I don't find myself enough capable to take this decision, and as you are the one with more experience with go packaging

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+
+ifeq ($(DEBUG), 0)
+	LDFLAGS += -s -w
+	GO_BUILD = CGO_ENABLED=0 go build -ldflags "$(LDFLAGS)" -trimpath
+else
+	GO_BUILD = CGO_ENABLED=0 go build -ldflags "$(LDFLAGS)"
+endif
 
 default: clean mod-tidy fmt vet-check test build
+debug: clean mod-tidy fmt vet-check build
```
Comment (stefanotorresi) [inline Makefile:14]: I guess this target can be left out now, right?

**Review by stefanotorresi — CHANGES_REQUESTED** [review]

**Review by stefanotorresi — APPROVED** [review]

**Pull request merged**